src/Compiler.ts

```diff
@@ -43,10 +43,27 @@ export function watchWithOptions(fileNames: string[], options: CompilerOptions):
         host = ts.createWatchCompilerHost(fileNames, options, ts.sys, ts.createSemanticDiagnosticsBuilderProgram);
     }
 
+    let fullRecompile = true;
     host.afterProgramCreate = program => {
         const transpiler = new LuaTranspiler(program.getProgram());
 
-        const status = transpiler.emitFilesAndReportErrors();
+        let status = transpiler.reportErrors();
+
+        if (status === 0) {
+            if (fullRecompile) {
+                status = transpiler.emitFilesAndReportErrors();
+            } else {
+                while (true) {
+                    const currentFile = program.getSemanticDiagnosticsOfNextAffectedFile();
+                    if (!currentFile) { break; }
+                    const fileStatus = transpiler.emitSourceFile(currentFile.affected as ts.SourceFile);
```

Comment:
currentFile has type `ts.SourceFile | ts.Program`, you should explicitly handle both cases using `ts.isSourceFile(currentFile)`, and `program.getSourceFiles()` to transpile all sourceFiles in a program, in case one is returned.